### PR TITLE
Make it possible to configure the default credentials cache

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
This is for https://github.com/awslabs/aws-sdk-rust/issues/462. It is just making it possible to change settings on the lazy caching builder used by the default credentials provider builder.

This should enable customizing the credential load timeout with:
```rust
let default_provider = DefaultCredentialsChain::builder()
    .load_timeout(Duration::from_secs(20))
    .build();
```

## Checklist
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/rust-runtime/aws-config/src/default_provider/credentials.rs [hunk 1]
```diff
@@ -114,6 +114,71 @@ impl Builder {
         self
     }
 
+    /// Timeout for the given [`ProvideCredentials`] implementation.
```
Comment (rcoh) [inline aws/rust-runtime/aws-config/src/default_provider/credentials.rs:117]: should this say "default credentials chain"? We should clarify that this is the entire chain and not an individual provider (but maybe we should actually add that too...)

File: aws/rust-runtime/aws-config/src/default_provider/credentials.rs [hunk 2]
```diff
@@ -114,6 +114,71 @@ impl Builder {
...
+    pub fn set_load_timeout(&mut self, timeout: Option<Duration>) -> &mut Self {
+        self.credential_cache.set_load_timeout(timeout);
+        self
+    }
+
+    /// Amount of time before the actual credential expiration time
+    /// where credentials are considered expired.
+    ///
+    /// For example, if credentials are expiring in 15 minutes, and the buffer time is 10 seconds,
+    /// then any requests made after 14 minutes and 50 seconds will load new credentials.
```
Comment (rcoh) [inline aws/rust-runtime/aws-config/src/default_provider/credentials.rs:137]: should update to have a first line that can stand alone

**Review by rcoh — APPROVED** [review]
LGTM, just some proposed docs improvements.

File: aws/rust-runtime/aws-config/src/default_provider/credentials.rs (hunk 1 above)
Comment (jdisanti) [reply aws/rust-runtime/aws-config/src/default_provider/credentials.rs:117]: Good call. Fixed.

**Pull request merged**